packages/datetime/src/dateRangePicker.tsx

```diff
@@ -417,56 +415,53 @@ export class DateRangePicker
-        if (compareDisplayMonth(potentialLeftDisplay, potentialRightDisplay) !== DisplayMonthOrder.IN_ORDER) {
-            potentialRightDisplay = getNextMonth(potentialLeftDisplay);
+        let potentialRightDisplay = this.state.rightDisplayMonth.clone();
+        if (!potentialLeftDisplay.isBefore(potentialRightDisplay)) {
+            potentialRightDisplay = potentialLeftDisplay.getNextMonth();
         }
 
         this.setDisplay(potentialLeftDisplay, potentialRightDisplay);
     }
 
     private handleRightYearSelectChange = (rightDisplayYear: number) => {
-        let potentialRightDisplay: DisplayMonth = [this.state.rightDisplayMonth, rightDisplayYear];
+        let potentialRightDisplay = new DisplayMonth(this.state.rightDisplayMonth.getMonth(), rightDisplayYear);
         const { minDate, maxDate } = this.props;
-        const adjustedMinDate = getDateNextMonth(minDate);
+        const adjustedMinDate = DateUtils.getDateNextMonth(minDate);
 
-        const minDisplayMonth: DisplayMonth = [adjustedMinDate.getMonth(), adjustedMinDate.getFullYear()];
-        const maxDisplayMonth: DisplayMonth = [maxDate.getMonth(), maxDate.getFullYear()];
+        const minDisplayMonth: DisplayMonth
+            = new DisplayMonth(adjustedMinDate.getMonth(), adjustedMinDate.getFullYear());
+        const maxDisplayMonth: DisplayMonth = new DisplayMonth(maxDate.getMonth(), maxDate.getFullYear());
 
-        if (compareDisplayMonth(minDisplayMonth, potentialRightDisplay) === DisplayMonthOrder.NOT_IN_ORDER) {
+        if (potentialRightDisplay.isBefore(minDisplayMonth)) {
             potentialRightDisplay = minDisplayMonth;
-        } else if (compareDisplayMonth(potentialRightDisplay, maxDisplayMonth) === DisplayMonthOrder.NOT_IN_ORDER) {
+        } else if (potentialRightDisplay.isAfter(maxDisplayMonth)) {
             potentialRightDisplay = maxDisplayMonth;
         }
 
-        let potentialLeftDisplay: DisplayMonth = [this.state.leftDisplayMonth, this.state.leftDisplayYear];
-        if (compareDisplayMonth(potentialLeftDisplay, potentialRightDisplay) !== DisplayMonthOrder.IN_ORDER) {
-            potentialLeftDisplay = getPreviousMonth(potentialRightDisplay);
+        let potentialLeftDisplay = this.state.leftDisplayMonth.clone();
+        if (!potentialRightDisplay.isAfter(potentialLeftDisplay)) {
+            potentialLeftDisplay = potentialRightDisplay.getPreviousMonth();
         }
 
         this.setDisplay(potentialLeftDisplay, potentialRightDisplay);
     }
 
-    private setDisplay(leftDisplay: DisplayMonth, rightDisplay: DisplayMonth) {
-        this.setState({
-            leftDisplayMonth: leftDisplay[0],
-            leftDisplayYear: leftDisplay[1],
-            rightDisplayMonth: rightDisplay[0],
-            rightDisplayYear: rightDisplay[1],
-        });
+    private setDisplay(leftDisplayMonth: DisplayMonth, rightDisplayMonth: DisplayMonth) {
```

Comment:
It's useful mainly so the `setState` in the other methods don't become a pain to read because they're named `potentialXXXX`. It's also nice to limit the places we have `setState` in this complicated component.